pyrovision/models/rexnet.py

```diff
@@ -0,0 +1,153 @@
+# Copyright (C) 2021, Pyronear contributors.
+
+# This program is licensed under the GNU Affero General Public License version 3.
+# See LICENSE or go to <https://www.gnu.org/licenses/agpl-3.0.txt> for full license details.
+
+from holocron.models.rexnet import ReXNet
+from typing import Dict, Any
+from pyrovision.models.utils import cnn_model
+from holocron.models.utils import load_pretrained_params
+
+__all__ = ['rexnet1_0x', 'rexnet1_3x', 'rexnet1_5x', 'rexnet2_0x', 'rexnet2_2x']
+
+
+default_cfgs: Dict[str, Dict[str, Any]] = {
+    'rexnet1_0x': {'width_mult': 1.0, 'depth_mult': 1.0,
+                   'url': 'https://github.com/frgfm/Holocron/releases/download/v0.1.2/rexnet1_0x_224-ab7b9733.pth'},
+    'rexnet1_3x': {'width_mult': 1.3, 'depth_mult': 1.0,
+                   'url': 'https://github.com/frgfm/Holocron/releases/download/v0.1.2/rexnet1_3x_224-95479104.pth'},
+    'rexnet1_5x': {'width_mult': 1.5, 'depth_mult': 1.0,
+                   'url': 'https://github.com/frgfm/Holocron/releases/download/v0.1.2/rexnet1_5x_224-c42a16ac.pth'},
+    'rexnet2_0x': {'width_mult': 2.0, 'depth_mult': 1.0,
+                   'url': 'https://github.com/frgfm/Holocron/releases/download/v0.1.2/rexnet2_0x_224-c8802402.pth'},
+    'rexnet2_2x': {'width_mult': 2.2, 'depth_mult': 1.0,
+                   'url': None},
+}
+
+model_urls = {
+    'rexnet1_0x': "https://github.com/pyronear/pyro-vision/releases/download/v0.1.0/rexnet1_0x_acp_2e017f83.pth",
+    'rexnet1_3x': None,
+    'rexnet1_5x': None,
+    'rexnet2_0x': None,
+    'rexnet2_2x': None
+}
+
+model_cut = -2
+
+
+def _rexnet(arch, pretrained=False, progress=True,
+            imagenet_pretrained=False, num_classes=1, lin_features=512,
+            dropout_prob=0.5, bn_final=False, concat_pool=True, **kwargs):
+
+    # Model creation
+    base_model = ReXNet(default_cfgs[arch]['width_mult'], default_cfgs[arch]['depth_mult'], **kwargs)
+
+    # Imagenet pretraining
+    if imagenet_pretrained:
+        if pretrained:
+            raise ValueError('imagenet_pretrained cannot be set to True if pretrained=True')
+
+        load_pretrained_params(base_model, default_cfgs[arch]['url'], progress)
```

Comment:
That section could be removed so that we pass the "imagenet_pretrained" value to the rexnet functions from Holocron